apps/web/src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -198,9 +198,13 @@ const useMessagePreviews = () => {
       }
     };
 
-    listConversations();
-    streamConversations();
-    streamAllMessages();
+    const setupConversationsList = async () => {
+      await listConversations();
+      await streamConversations();
```

Comment:
Wait. Neither `streamConversations` or `streamAllMessages` ever return. This will stop at line 203 and never make it to line 204

Reply:
You should be able to confirm by seeing if new messages are being streamed in for anything other than the currently open conversation.

Reply:
Great catch. It should be the `list()` method that was causing trouble anyways. I'll reconfirm and add timing benchmarks to the description. It still should be significantly faster as-is still (just broken streaming).